### ISSUE TITLE
Pillow v3+ breaks Fit-In + rotation: incorrect crop and black background fill

### ISSUE THREAD

**Issue by thewhit** [thumbor/thumbor#725]
### Thumbor request URL
http://localhost/unsafe/fit-in/1500x844/left/top/smart/filters:rotate(-90)/http://cloud.newcityarts.org/er20tpxsv6oxijm4xcafc4ol96havgoygo_frhg9vvs.JPG
![er20tpxsv6oxijm4xcafc4ol96havgoygo_frhg9vvs-1](https://cloud.githubusercontent.com/assets/636353/15633681/c74bc03e-2581-11e6-9c5d-6b0cbb9202b1.JPG)

### Expected behaviour
The image should be rotated and then cropped to fit in the area, like this example with Pillow 2.7.0: https://localhost/unsafe/fit-in/1500x844/left/top/smart/filters:rotate(-90)/http://cloud.newcityarts.org/er20tpxsv6oxijm4xcafc4ol96havgoygo_frhg9vvs.JPG.
![er20tpxsv6oxijm4xcafc4ol96havgoygo_frhg9vvs](https://cloud.githubusercontent.com/assets/636353/15633683/ca792a12-2581-11e6-9fd4-ad38fd8c9d45.JPG)

### Actual behaviour
The image is cropped, rotated, and the extra area is filled in with black.

### Operating system
Ubuntu 12.04 LTS

### Your thumbor.conf
https://gist.github.com/thewhit/3176318f87f7129172f9061b9ba4ffdb

**Comment by hhankj2u** [comment]
I have a similar situation when turn on **RESPECT_ORIENTATION**
After some research, I found [this](http://stackoverflow.com/questions/5252170/specify-image-filling-color-when-rotating-in-python-with-pil-and-setting-expand).
Apply for thumbor:
Open file **/path/to/python2.7/site-packages/thumbor/engines/extensions/pil.py** and edit function **rotate** (line 107). Replace original
```
def rotate(self, degrees):
        # PIL rotates counter clockwise
        self.image = self.image.rotate(degrees)
```
to this:
```
def rotate(self, degrees):
        # PIL rotates counter clockwise
        # converted to have an alpha layer
        tmp_img = self.image.convert('RGBA')
        # rotated image
        rotate_img = tmp_img.rotate(degrees)
        # a white image same size as rotated image
        fff = Image.new('RGBA', rotate_img.size, (255,)*4)
        # create a composite image using the alpha layer of rotate_img as a mask
        self.image = Image.composite(rotate_img, fff, rotate_img)
        #self.image = self.image.rotate(degrees)
```

**Comment by guilhermef** [comment]
It's fixed on 6.0.2
